src/stakingSetup.js

```diff
@@ -0,0 +1,153 @@
+        log.info("Transfering cennz to " + account.meta.name)            
+            const cennz = await api.tx.genericAsset.transfer(16000, account.address, "100_000_000_000_000")
+            .signAndSend(acliceKeyring, (result) => {
+                log.info(`Current status is ${result.status}`);
+
+                if (result.isCompleted) {
+                    if (result.isFinalized) {
+                        hash = result.status.asFinalized;
+                    }
+                    else { // error
+                        log.info("Transaction Rejected");
+                        throw [APP_FAIL_TRANSACTION_REJECTED, `Transaction rejected ${result.status} for ${account.meta.name}`];
+                    }
+                    cennz();
+                }
+            })
+            .catch(err => transaction_error = err);
+            await sleep();
+            
+    }
+
+    for (account of stashAccount) {
+        log.info("SETTING UP FOR CENTRAPAY -------------- \n Accout raw seed: " + account.meta.name)
+        log.info(`Transfering centrapay to ${account.meta.name} and address ${account.address}`)
+        const unsub = await api.tx.genericAsset.transfer(16001, account.address, "100_000_000_000_000")
+            .signAndSend(acliceKeyring, (result) => {
+                log.info(`Current status is ${result.status}`);
+
+                if (result.isCompleted) {
+                    if (result.isFinalized) {
+                        hash = result.status.asFinalized;
+                    }
+                    else { // error
+                        log.info("Transaction Rejected");
+                        throw [APP_FAIL_TRANSACTION_REJECTED, `Transaction rejected ${result.status}`];
+                    }
+                    unsub();
+                }
+            })
+            .catch(err => transaction_error = err);
+            await sleep();
+    }
+
+    await bondStashAccounts(api, stashAccount);
+    await setSessionKey(api);
+    await setNodesValidating(api);
+}
+
+function sleep(){
+    return new Promise(resolve=>{
+        setTimeout(resolve,DEFAULT_SLEEP)
+    })
+}
+
+async function bondStashAccounts(api, stashes){
+    
+    for(i=0; i< controllerAddrs.length; i++){
+       let transfer = api.tx.staking.bond(controllerAddrs[i], "5_000_000", 0);
+        // Sign and Send the transaction
+        transfer.signAndSend(stashes[i], ({ events = [], status, type }) => {
+            if (type === 'Finalised') {
+                console.log('Successful transfer ' + ' with hash ' + status.asFinalised.toHex());
+              } else {
+                console.log('Status of transfer: ' + type);
+              }
+              events.forEach(({ phase, event: { data, method, section } }) => {
+                console.log(phase.toString() + ' : ' + section + '.' + method + ' ' + data.toString());
+              });
+        })
+        .catch(err => transaction_error = err);
+
+        await sleep();
```

Comment:
This is another race condition, also on a 6 second block chain, this will occasionally fail to sleep long enough